File: reddit-api-spring/src/main/java/com/gusrylmubarok/reddit/backend/dto/VoteDto.java
package com.gusrylmubarok.reddit.backend.dto;

import com.gusrylmubarok.reddit.backend.model.VoteType;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class VoteDto {
    private VoteType voteType;
    private Long postId;
}

File: src/main/kotlin/com/reddot/data/model/VoteDTO.kt
package com.reddot.data.model

import com.fasterxml.jackson.annotation.JsonProperty
import com.reddot.data.enumeration.VoteType
import jakarta.validation.constraints.NotEmpty

data class VoteDTO(
    @field:NotEmpty
    @JsonProperty("vote_type")
    val voteType: VoteType,

    @field:NotEmpty
    @JsonProperty("post_id")
    val postId: Long
)